packages/contracts/docs/osx/02-how-to-guides/02-plugin-development/04-upgradeable-plugin/04-subsequent-builds.md

```diff
@@ -0,0 +1,141 @@
+---
+title: Creating Subsequent Builds for Upgradeable Plugins
+---
+
+## How to create a subsequent build to an Upgradeable Plugin
+
+A build is a new implementation of your Upgradeable Plugin. Upgradeable contracts offer advantages because you can cheaply change or fix the logic of your contract without losing the storage of your contract.
+
+In this how to guide, we'll go through how we can create these builds for our plugins. Specifically, we'll showcase two specific types of builds - one that modifies the storage of the plugins, another one which modifies its bytecode. Both are possible and can be implemented within the same build implementation as well.
+
+### 1. Make sure your previous build is deployed and published
+
+Make sure you have at least one build already deployed and published into the Aragon protocol. Make sure to check out our [publishing guide](../07-publication/index.md) to ensure this step is done.
+
+### 2. Create a build implementation modifying storage
+
+In this second build, we want to change certain sections of our existing plugin implementation - we want to update the storage of our plugin with new values. In this specific case, we will add a second storage variable `address public account;`. Additional to the `initializeFromBuild2` function, we also want to add a second setter function `storeAccount` that uses the same permission as `storeNumber`.
+
+As you can see, we're still inheritting from the `PluginUUPSUpgradeable` contract and simply overriding some implementation from the previous build. The idea is that when someone upgrades the plugin and calls on these functions, they'll use this new upgraded implementation, rather than the older one.
+
+```solidity
+import {PluginUUPSUpgradeable, IDAO} '@aragon/osx/core/plugin/PluginUUPSUpgradeable.sol';
+
+/// @title SimpleStorage build 2
+contract SimpleStorageBuild2 is PluginUUPSUpgradeable {
+  bytes32 public constant STORE_PERMISSION_ID = keccak256('STORE_PERMISSION');
+
+  uint256 public number; // added in build 1
+  address public account; // added in build 2
+
+  /// @notice Initializes the plugin when build 2 is installed.
+  function initializeBuild2(
+    IDAO _dao,
+    uint256 _number,
+    address _account
+  ) external reinitializer(2) {
```

Comment:
We should mention that `reinitializer(2)` 's number needs to be incremented at each new contracts build

Reply:
Simple storage plugin update - added to notes!